src/main.rs

```diff
@@ -27,35 +28,36 @@ run()
 
 fn main() {
     // init device
-    let device_handle = devices::tactrix_openport::new();
+    let device_handle = Arc::new(devices::tactrix_openport::new());
     // init server
     let server = std::net::TcpListener::bind("127.0.0.1:9001").unwrap();
     // listen for connections
-    let websockets = Arc::new(Mutex::new(vec![]));
+    let mut websockets = vec![];
     for stream in server.incoming() {
-        let mut websocket = tungstenite::server::accept(stream.unwrap()).unwrap();
-        websockets.lock().unwrap().push(websocket);
+        let websocket = Arc::new(Mutex::new(tungstenite::server::accept(stream.unwrap()).unwrap()));
+        websockets.push(websocket.clone());
+        let handle = device_handle.clone();
```

Comment:
we explicitly clone the device handle and use the cloned handle in the body of the closure. This increments the reference count, and allows the original handle to be cloned from in future invocations.